docs/path-resolution.rst

```diff
@@ -0,0 +1,465 @@
+    is not subject to the normalization rules you would typically expect from filesystem paths.
+    Any ``/./`` or ``/../`` seguments or sequences of multiple slashes remain a part of it.
+    When the source is provided via Standard JSON interface it is entirely possible to associate
+    different content with source unit names that would refer to the same file on disk.
+
+When the source is not available in the virtual filesystem, the compiler passes the source unit name
+to the import callback.
+The Host Filesystem Loader will attempt to use it as a path and look up the file on disk.
+At this point the platform-specific normalization rules kick in and ``/project/lib/math.sol`` and
+``/project/lib/../lib///math.sol`` may actually result in the same file being loaded.
+Note, however, that the compiler will still see them as separate source units that just happen to
+have identical content.
+
+.. index:: ! relative import, ! import; relative
+.. _relative-imports:
+
+Relative Imports
+----------------
+
+An import starting with ``./`` or ``../`` is a *relative import*.
+Such imports specify a path relative to the source unit name of the importing source unit:
+
+.. code-block:: solidity
+    :caption: /project/lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: /project/lib/util.sol
+    import "../token.sol" as token; // source unit name: /project/token.sol
+
+.. code-block:: solidity
+    :caption: lib/math.sol
+
+    import "./util.sol" as util;    // source unit name: lib/util.sol
+    import "../token.sol" as token; // source unit name: token.sol
+
+.. note::
+
+    Note the difference between a *relative import* (e.g. ``import "./util.sol"``) and a
+    *direct import with a relative path* (e.g. ``import "util.sol"``).
+    It is a relative import only if it starts with ``./`` or ``../``.
+    Otherwise it is a direct import.
+
+The compiler computes a source unit name from the import path in the following way:
+
+1. First a prefix is computed
+
+    - Prefix is initialized with the source unit name of the importing source unit.
+    - The last segment is removed from the prefix.
+    - Then, for every leading ``../`` segment in the normalized import path the last segment is repeatedly removed from the prefix.
+
+2. Then the prefix is prepended to the normalized import path.
+   If the prefix is non-empty, a single slash is inserted between it and the import path.
+
+In the above the removal of the last segment is performed as follows:
+
+1. Everything past the last slash is removed (i.e. ``a/b//c.sol`` becomes ``a/b//``).
+2. All trailing slashes are removed (i.e. ``a/b//`` becomes ``a/b``).
+
+The normalization rules are the same as for UNIX paths, namely:
+
+- All the internal ``./`` segments are removed.
+- Every internal ``../``  segment backtracks one level up in the hierarchy.
+- Multiple slashes are squashed into a single one.
+
+Note that normalization is performed only on the import path.
```

Comment:
Maybe we could emphasize that this only applies for relative import paths. E.g. `Note that normalization is performed only on *relative* import paths.`

Reply:
My emphasis here was actually on the fact that it's an import path as opposed to the source unit name of the importing module or source unit name resulting from the import. I think that mentioning that it's relative here would detract from that. And the whole section is about relative imports anyway.